src/components/DisciplineListItem.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { ReactElement } from 'react'
+
+import { Discipline } from '../constants/endpoints'
+import { childrenDescription, childrenLabel } from '../services/helpers'
+import ListItem from './ListItem'
+
+interface DisciplineListItemProps {
+  item: Discipline
+  onPress: () => void
+  badge: boolean
+}
+
+const DisciplineListItem = ({ item, onPress, badge }: DisciplineListItemProps): ReactElement | null => {
+  const { numberOfChildren, title, icon } = item
+  const badgeLabel = badge ? numberOfChildren.toString() : undefined
+  const description = badge ? childrenLabel(item) : childrenDescription(item)
```

Comment:
I think this is somehow confusing now.
description either contains the numberOfChildren plus the "unit" (e.g. Modul) or just contains the "unit". Maybe you can think of a solution to make this more clear.